sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -72,57 +72,76 @@ case class CurrentTimestamp() extends LeafExpression with CodegenFallback {
 }
 
 /**
- * Adds a number of days to startdate.
+ * Adds a number of days to date/timestamp.
  */
 @ExpressionDescription(
-  usage = "_FUNC_(start_date, num_days) - Returns the date that is num_days after start_date.",
+  usage = "_FUNC_(instant, num_days) - Returns the date/timestamp that is num_days after instant.",
   extended = "> SELECT _FUNC_('2016-07-30', 1);\n '2016-07-31'")
-case class DateAdd(startDate: Expression, days: Expression)
+case class DateAdd(instant: Expression, days: Expression)
```

Comment:
`DATE_ADD` is the name of this function in quite a lot of SQL databases (Hive, MySql & SqlServer to name a few). 

We could name the expression `AddDays`, and expose it as `DATE_ADD`.